### ISSUE TITLE
gcp-compute-subnet asset names are not unique enough

### ISSUE THREAD

**Issue by tas50** [mondoohq/cnquery#1507]
When scanning a GCP project with `--discover all` you get a large number of gcp-compute-subnet assets named "default". There's no way to distinguish between these. We need a more specific name.

```
 ~/d/cnquery   v8 $  cnspec scan gcp project tim-learning-project --discover all
→ loaded configuration from /Users/tsmith/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=45
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default                               ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default-allow-icmp                    ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default-allow-internal                ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default-allow-rdp                     ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U
 default-allow-ssh                     ────────────────────────────────────────────────────────────────────────────────────────────────────────  n/a score: U

 1/45 scanned 44/45 n/a                ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%

Asset: GCP project tim-learning-project
---------------------------------------

Data queries:
asset.title: "GCP Project"
gcp.compute.instances.where: []
gcp.project.id: "tim-learning-project"
gcp.project.createTime: 2022-11-14 13:25:47.996 -0800 PST
gcp.compute.networks: [
  0: gcp.project.computeService.network name="default"
]
name: "Tim Learning Project"
id: "tim-learning-project"
state: "ACTIVE"
labels: {}
gcp.project: {
  labels: {}
  id: "tim-learning-project"
  state: "ACTIVE"
  name: "Tim Learning Project"
... 2 more lines ...
gcp.project.gke.clusters: []
gcp.project.gke.clusters.length: 0
gcp.project.iamPolicy.where: []
gcp.compute.instances.where.length: 0
gcp.project.state: "ACTIVE"
gcp.compute.networks.length: 1
gcp.project.labels: {}
gcp.compute.instances.where: []
mondoo.version: "8.23.0"
gcp.project.services.where: [
  0: gcp.service name="bigquery.googleapis.com"
  1: gcp.service name="bigquerymigration.googleapis.com"
  2: gcp.service name="bigquerystorage.googleapis.com"
  3: gcp.service name="cloudapis.googleapis.com"
  4: gcp.service name="cloudtrace.googleapis.com"
  5: gcp.service name="compute.googleapis.com"
  6: gcp.service name="datastore.googleapis.com"
  7: gcp.service name="logging.googleapis.com"
  8: gcp.service name="monitoring.googleapis.com"
... 8 more lines ...
gcp.project.name: "Tim Learning Project"
gcp.project.iamPolicy: [
  0: {
    role: "roles/compute.serviceAgent"
  }
  1: {
    role: "roles/editor"
  }
  2: {
    role: "roles/owner"
  }
... 1 more lines ...
gcp.project.iamPolicy.where: [
  0: gcp.resourcemanager.binding id = tim-learning-project-2
]

Checks:
✓ Pass:  A 100  Ensure instances are not configured to use the default service account with full access to all Cloud APIs
✓ Pass:  A 100  Ensure that Cloud Storage buckets are not anonymously or publicly accessible
✓ Pass:  A 100  Ensure "Block Project-wide SSH keys" is enabled for VM instances
✓ Pass:  A 100  Ensure oslogin is enabled for compute instances
✓ Pass:  A 100  Ensure that Cloud Storage buckets have uniform bucket-level access enabled
✓ Pass:  A 100  Ensure that instances are not configured to use the default service account

Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Network"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
asset.title: "GCP Compute Subnetwork"
mondoo.version: "8.23.0"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default
--------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Subnetwork"


Asset: default-allow-icmp
-------------------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Firewall"


Asset: default-allow-internal
-----------------------------

Data queries:
asset.title: "GCP Compute Firewall"
mondoo.version: "8.23.0"


Asset: default-allow-rdp
------------------------

Data queries:
mondoo.version: "8.23.0"
asset.title: "GCP Compute Firewall"


Asset: default-allow-ssh
------------------------

Data queries:
asset.title: "GCP Compute Firewall"
mondoo.version: "8.23.0"



Scanned 45 assets

GCP Compute Firewall
    U default-allow-icmp
    U default-allow-internal
    U default-allow-rdp
    U default-allow-ssh

GCP Compute Network
    U default

GCP Compute Subnetwork
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default
    U default

GCP Project
    A GCP project tim-learning-project

Summary
=======

Score Distribution		Asset Distribution
------------------		------------------
A   1 assets      		GCP Compute Network      1
B   0 assets      		GCP Compute Firewall     4
C   0 assets      		GCP Project              1
D   0 assets      		GCP Compute Subnetwork  39
F   0 assets
U  44 assets
```

**Comment by vjeffrey** [comment]
hmmm, is "default" what the subnetwork is named in gcp? what does it show there?

**Comment by tas50** [comment]
Yeah they're all called "default", which makes the name a not so great identifier on its own.

**Comment by vjeffrey** [comment]
do u have any ideas of how to enhance the name? this is also the name that shows up in the ui. for ecs containers i've done things like name-ip. would name-location be enough for this case?

**Comment by tas50** [comment]
Location would probably do it since they have to be unique in the zone

**Comment by tas50** [comment]
Fixed in 8.24. Thanks @vjeffrey
